### PR TITLE
CRM-19964 Event registration creates contribution without line item

### PR DESCRIPTION
**Pull request by pradpnayak**
* [CRM-19964: Event registration creates contribution without line item](https://issues.civicrm.org/jira/browse/CRM-19964)

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins test this please

**Comment by Edzelopez** [comment]
@eileenmcnaughton I tried creating a unit test for this, but it seems like there isn't a way to emulate a form submission for an online registration similar to the "$form->testSubmit" we use for contribution pages. I also can't find examples of online registration test cases under tests/phpunit/api/v3/.

Hasn't a helper function been created yet or is there another way I should be doing this?

**Comment by eileenmcnaughton** [comment]
Hey @Edzelopez that's the sort of question I like :-)

I only added the first Event form test recently - look for this class CRM_Event_Form_Registration_ConfirmTest

**Comment by Edzelopez** [comment]
Added unit test

File: CRM/Event/Form/Registration/Register.php [hunk 1]
```diff
@@ -1015,6 +1015,7 @@ public function postProcess() {
         else {
           $submittedLineItems = array($lineItem);
         }
+        $submittedLineItems = array_filter($submittedLineItems);
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/Registration/Register.php]: what is the reason for this?

**Comment by eileenmcnaughton** [comment]
I'm going to merge this - the use of array_filter feels like hack to get around another problem. OTOH it is locked in with a unit test & fixes an important bug.

**Pull request merged**